src/kvstore.c

```diff
@@ -778,3 +782,17 @@ int kvstoreDictDelete(kvstore *kvs, int didx, const void *key) {
     }
     return ret;
 }
+
+/* Reallocates the memory used by the dict list in kvstore using the provided
+ * allocation function. This feature was added for the active defrag feature.
+ *
+ * The 'defragfns' callbacks are called with a reference to a pointer
+ * that callback can reallocate. */
+void kvstoreScanDefrag(kvstore *kvs, kvstoreDefragFunction *defragfn) {
```

Comment:
The term "scan" is a bit odd (considering our dictScan mechanism). And also I think it's not specific for "defrag"? (Other than the fact it passes a ref and expects it to be modified).

Reply:
yeah, I've been struggling with that name for a long time.
what about `kvstoreDefragDicts`?
i think adding `defrag` is still needed, to avoid this method being abused or misused.

Reply:
the name you suggested is better, but considering it doesn't defrag the keys and values (we have `kvstoreDictScanDefrag` for that), maybe `kvstoreDefragRoot`, or `kvstoreDefragDictLUT`?